### PR TITLE
[IMP] *: add v18 digest tips

### PR DESCRIPTION
**Pull request by amdi-odoo**
Add more digest tips to showcase the new features added for v18.
Tips sequence numbers going from 3100 to 4400.

Technical:
Had to slightly change the name of a company in the hr_holidays tests
to be different from the base test company declared in the module common test file.
Adding the digest module as dependency to the hr module loads helpdesk before hr_holidays.
An helpdesk team is thus created each time a company is created and using the same company
name generates the same mail alias which breaks the alias unicity check.

Same thing happens in the hr_work_entry_holidays tests, the modules loading order has slightly changed
following the addition of the digest dependency in hr. This causes a slight increase in the query count
when testing the validation of a leave.
Ideally those tests should become post_install in order to be more deterministic.

Related: https://github.com/odoo/enterprise/pull/67244

Task-3762938

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Review by Odoonan — COMMENTED** [review]
Hello @amdi-odoo 

LGTM, I see there is a conflict though :)
Does the hr_work_entry_holidays test update is also due to a new module loading order?

Side note : last max sequence was 3000 and you added tips with 3100 -> 4400, can you just specify the range or max sequence in the commit message? This way next time we can easily have the info when adding some tips :)

(pps: 'dependendy' typo in commit message)

Cheers and thanks for the work!

**Comment by amdi-odoo** [comment]
Hello @Odoonan :slightly_smiling_face: !
Yes the `hr_work_entry_holidays` test update is also a consequence to adding the digest module as dependency to hr :slightly_smiling_face: 
Thx for the early validation! :smile: 
Cheers!

File: addons/account/data/digest_data.xml [hunk 1]
```diff
@@ -16,6 +16,19 @@
...
+
+        <record id="digest_tip_account_1" model="digest.tip">
+            <field name="name">Tip: Stop chasing the Documents you need</field>
+            <field name="sequence">3100</field>
+            <field name="group_id" ref="account.group_account_invoice"/>
+            <field name="tip_description" type="html">
+<div>
+    <p class="tip_title">Tip: Stop chasing the Documents you need</p>
+    <p class="tip_content">Missing a document for a banking statement? Use the Documents apps to request it and let the owner upload it at the right place.</p>
+    <img src="/digest/static/src/img-v17/request-document.gif" width="540" class="illustration_border"/>
```
Comment (tde-banana-odoo) [inline addons/account/data/digest_data.xml:31]: Would name 18-invoice-request-document.gif

File: addons/base_automation/data/digest_data.xml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    <data>
+        <record id="digest_tip_base_automation_0" model="digest.tip">
+            <field name="name">Tip: Automate everything with Automation Rules</field>
+            <field name="sequence">3700</field>
+            <field name="group_id" ref="base.group_system"/>
+            <field name="tip_description" type="html">
+<div>
+    <p class="tip_title">Tip: Automate everything with Automation Rules</p>
+    <p class="tip_content">Send an email when an object changes state, archive records after a month of inactivity or remind yourself to follow-up on tasks when a specific tag is added.<br/>With Automation Rules, you can automate any workflow.</p>
+    <img src="/digest/static/src/img-v17/automation-rules.gif" width="540" class="illustration_border"/>
```
Comment (tde-banana-odoo) [inline addons/base_automation/data/digest_data.xml:12]: 18-automation-rules.gif

File: addons/crm/data/digest_data.xml [hunk 3]
```diff
@@ -83,6 +83,30 @@
...
+        </record>
+        <record id="digest_tip_crm_6" model="digest.tip">
+            <field name="name">Tip: Identify Bottlenecks at a glance</field>
+            <field name="sequence">3600</field>
+            <field name="group_id" ref="sales_team.group_sale_salesman"/>
+            <field name="tip_description" type="html">
+<div>
+    <p class="tip_title">Tip: Identify Bottlenecks at a glance</p>
+    <p class="tip_content">Check the Stage Tracker of Leads to identify bottlenecks in your Sales process.</p>
+    <img src="/digest/static/src/img-v17/lead-stages-bottleneck.png" width="540" class="illustration_border"/>
```
Comment (tde-banana-odoo) [inline addons/crm/data/digest_data.xml:97]: 18-lead-stage-bottleneck.png

File: addons/crm/data/digest_data.xml [hunk 4]
```diff
@@ -83,6 +83,30 @@
...
+        </record>
+        <record id="digest_tip_crm_7" model="digest.tip">
+            <field name="name">Tip: Turn Sales Forecasting into Child's Play</field>
+            <field name="sequence">4400</field>
+            <field name="group_id" ref="sales_team.group_sale_salesman"/>
+            <field name="tip_description" type="html">
+<div>
+    <p class="tip_title">Tip: Turn Sales Forecasting into Child's Play</p>
+    <p class="tip_content">Use the CRM Forecast Kanban to easily define when Opportunities are expected to be won.</p>
+    <img src="/digest/static/src/img-v17/forecast-opportunities.gif" width="540" class="illustration_border"/>
```
Comment (tde-banana-odoo) [inline addons/crm/data/digest_data.xml:109]: 18-opportunity-forecast.gif

File: addons/digest/data/digest_tips_data.xml [hunk 5]
```diff
@@ -69,6 +69,42 @@
...
+        </record>
+        <record id="digest_tip_digest_5" model="digest.tip">
+            <field name="name">Tip: Join the Dark Side</field>
+            <field name="sequence">3300</field>
+            <field name="group_id" ref="base.group_user"/>
+            <field name="tip_description" type="html">
+<div>
+    <p class="tip_title">Tip: Join the Dark Side</p>
+    <p class="tip_content">Feeling eye strain? Give your eyes a break by switching to Dark Mode.</p>
+    <img src="/digest/static/src/img-v17/switch-theme.gif" width="540" class="illustration_border"/>
```
Comment (tde-banana-odoo) [inline addons/digest/data/digest_tips_data.xml:83]: 18-web-dark-theme.gif

File: addons/digest/data/digest_tips_data.xml [hunk 6]
```diff
@@ -69,6 +69,42 @@
...
+            </field>
+        </record>
+        <record id="digest_tip_digest_6" model="digest.tip">
+            <field name="name">Tip: Your Business right in their pocket</field>
+            <field name="sequence">3400</field>
+            <field name="group_id" ref="base.group_system"/>
+            <field name="tip_description" type="html">
+<div>
+    <p class="tip_title">Tip: Your Business right in their pocket</p>
+    <p class="tip_content">Connect your WhatsApp account to Odoo to receive inbound messages directly in your CRM.</p>
```
Comment (tde-banana-odoo) [inline addons/digest/data/digest_tips_data.xml:94]: This one requires enterprise code, hence I suggest to not have it in community (I know, marketing blabla, but quand même :D ) .

File: addons/digest/data/digest_tips_data.xml [hunk 7]
```diff
@@ -69,6 +69,42 @@
...
+    <img src="/digest/static/src/img-v17/wa-message-flow.png" width="540" class="illustration_border"/>
```
Comment (tde-banana-odoo) [inline addons/digest/data/digest_tips_data.xml:95]: 18-whatsapp-message-flow.png

File: addons/digest/data/digest_tips_data.xml [hunk 8]
```diff
@@ -69,6 +69,42 @@
...
+        </record>
+        <record id="digest_tip_digest_7" model="digest.tip">
+            <field name="name">Tip: Personalize your Home Menu</field>
+            <field name="sequence">3800</field>
+            <field name="group_id" ref="base.group_user"/>
+            <field name="tip_description" type="html">
+<div>
+    <p class="tip_title">Tip: Personalize your Home Menu</p>
+    <p class="tip_content">Click and hold on a Menu Item to reorder your Apps to your liking.</p>
+    <img src="/digest/static/src/img-v17/change-apps-order.gif" width="540" class="illustration_border"/>
```
Comment (tde-banana-odoo) [inline addons/digest/data/digest_tips_data.xml:107]: 18-web-app-order.gif

File: addons/hr/data/digest_data.xml [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+    <data>
+        <record id="digest_tip_hr_0" model="digest.tip">
+            <field name="name">Tip: Where's Bryan?</field>
+            <field name="sequence">3500</field>
+            <field name="group_id" ref="hr.group_hr_manager"/>
+            <field name="tip_description" type="html">
+<div>
+    <p class="tip_title">Tip: Where's Bryan?</p>
+    <p class="tip_content">Activate Remote Work to let Employees specify where they are working from.</p>
+    <img src="/digest/static/src/img-v17/work-location.gif" width="540" class="illustration_border"/>
```
Comment (tde-banana-odoo) [inline addons/hr/data/digest_data.xml:12]: 18-hr-work-location.gif

File: addons/hr/data/digest_data.xml [hunk 10]
```diff
@@ -0,0 +1,17 @@
+<?xml version='1.0' encoding='utf-8'?>
+<odoo>
+    <data>
+        <record id="digest_tip_hr_0" model="digest.tip">
+            <field name="name">Tip: Where's Bryan?</field>
```
Comment (tde-banana-odoo) [inline addons/hr/data/digest_data.xml:5]: Suggestion more related to actual usage of the app
```suggestion
            <field name="name">Tip: Micro Manage Bryan. Oh pardon, micro guidance</field>
```

File: addons/hr_work_entry_holidays/tests/test_performance.py [hunk 11]
```diff
@@ -32,7 +32,7 @@ def test_performance_leave_validate(self):
         self.richard_emp.generate_work_entries(date(2018, 1, 1), date(2018, 1, 2))
         leave = self.create_leave(datetime(2018, 1, 1, 7, 0), datetime(2018, 1, 1, 18, 0))
 
-        with self.assertQueryCount(__system__=114, admin=115):  # com 96/97
+        with self.assertQueryCount(__system__=117, admin=118):  # com 96/97
```
Comment (tde-banana-odoo) [inline addons/hr_work_entry_holidays/tests/test_performance.py:35]: Mmmh ? Why this change ?

File: addons/project/data/digest_data.xml [hunk 12]
```diff
@@ -36,6 +36,32 @@
...
+
+        <record id="digest_tip_project_2" model="digest.tip">
+            <field name="name">Tip: Your Own Personal Kanban</field>
+            <field name="sequence">3200</field>
+            <field name="group_id" ref="project.group_project_user"/>
+            <field name="tip_description" type="html">
+<div>
+    <p class="tip_title">Tip: Your Own Personal Kanban</p>
+    <p class="tip_content">Use personal stages to organize your tasks and create your own workflow.</p>
+    <img src="/digest/static/src/img-v17/personal-stages.gif" width="540" class="illustration_border"/>
```
Comment (tde-banana-odoo) [inline addons/project/data/digest_data.xml:51]: 18-project-personal-stages.gif

File: addons/project/data/digest_data.xml [hunk 13]
```diff
@@ -36,6 +36,32 @@
...
+
+        <record id="digest_tip_project_3" model="digest.tip">
+            <field name="name">Tip: Project-Specific Fields</field>
+            <field name="sequence">3900</field>
+            <field name="group_id" ref="project.group_project_manager"/>
+            <field name="tip_description" type="html">
+<div>
+    <p class="tip_title">Tip: Project-Specific Fields</p>
+    <p class="tip_content">Add project-specific property fields on tasks to customize your project management process.</p>
+    <img src="/digest/static/src/img-v17/task-property-fields.gif" width="540" class="illustration_border"/>
```
Comment (tde-banana-odoo) [inline addons/project/data/digest_data.xml:64]: 18-project-property-fields.gif

File: addons/stock/data/digest_data.xml [hunk 14]
```diff
@@ -19,6 +19,18 @@
...
+        </record>
+        <record id="digest_tip_stock_1" model="digest.tip">
+            <field name="name">Tip: Monitor Lot details</field>
+            <field name="sequence">4000</field>
+            <field name="group_id" ref="stock.group_stock_user"/>
+            <field name="tip_description" type="html">
+<div>
+    <p class="tip_title">Tip: Monitor Lot details</p>
+    <p class="tip_content">Store and retrieve information regarding every Lot/Serial Number (condition, product info, ...).</p>
+    <img src="/digest/static/src/img-v17/lot-property-fields.gif" width="540" class="illustration_border"/>
```
Comment (tde-banana-odoo) [inline addons/stock/data/digest_data.xml:33]: 18-stock-property-fields.gif

**Review by tde-banana-odoo — COMMENTED** [review]
Thanks :)  suggested new names for images, based on {version}-{module/model}-{details} scheme, easier to follow I think. As it is the "18" official tips, I suggest to talk about v18 more than v17 ^^

Once updated I will ask to upload on cdn

File: addons/hr_work_entry_holidays/tests/test_performance.py (hunk 11 above)
Comment (amdi-odoo) [reply addons/hr_work_entry_holidays/tests/test_performance.py:35]: It is also a consequence to adding the digest module as dependency to hr :slightly_smiling_face: 
Similarly to the other test I updated, a specific module is now loaded before `hr_work_entry_holidays` because of the digest dependency in `hr` and the execution flow has slightly changed because of that.
I've been investigating for a few hours to know exactly what's happening but I can't seem to grasp which module is adding the queries. :confused: I always get a query count lower than 117 :upside_down_face:

Comment (tde-banana-odoo) [reply addons/hr_work_entry_holidays/tests/test_performance.py:35]: Arf, actually those tests should be made post_install, otherwise they depend on db state. So yes just update with an explanation that module loading changed, and that ideally those tests should become post_install in order to be more deterministic :)

File: addons/base_automation/__manifest__.py [hunk 15]
```diff
@@ -15,10 +15,11 @@
 Sales Team, or an opportunity which still has status pending after 14 days might
 trigger an automatic reminder email.
     """,
-    'depends': ['base', 'resource', 'mail', 'sms'],
+    'depends': ['base', 'digest', 'resource', 'mail', 'sms'],
```
Comment (kmagusiak) [inline addons/base_automation/__manifest__.py:18]: Instead of adding a dependency here, you could buid a module `digest_automation` which is auto-installable.

Comment (tde-banana-odoo) [reply addons/base_automation/__manifest__.py:18]: Would suggest to not add a bridge module just for that, as I heard we try to keep module number under control ^^ (each one slowing down registry, ...)

File: addons/base_automation/data/digest_data.xml [hunk 16]
```diff
@@ -0,0 +1,17 @@
...
+    <data>
+        <record id="digest_tip_base_automation_0" model="digest.tip">
+            <field name="name">Tip: Automate everything with Automation Rules</field>
+            <field name="sequence">3700</field>
+            <field name="group_id" ref="base.group_system"/>
+            <field name="tip_description" type="html">
+<div>
+    <p class="tip_title">Tip: Automate everything with Automation Rules</p>
+    <p class="tip_content">Send an email when an object changes state, archive records after a month of inactivity or remind yourself to follow-up on tasks when a specific tag is added.<br/>With Automation Rules, you can automate any workflow.</p>
+    <img src="/digest/static/src/img-v18/18-automation-rules.gif" width="540" class="illustration_border"/>
```
Comment (rco-odoo) [inline addons/base_automation/data/digest_data.xml:12]: Maybe I'm stupid, but the image you refer to is specific to this module.  Why put it in module `digest`?

Comment (amdi-odoo) [reply addons/base_automation/data/digest_data.xml:12]: All the gifs and images are temporarily set in a folder in digest as I'm waiting for them to be hosted on odoo cdn like all the other digest tips :slightly_smiling_face: 
Example: https://download.odoocdn.com/digests/crm/static/src/img/milk-pipeline-progress.gif
Once they're hosted all the `src`'s will be updated :smile:

File: addons/base_automation/__manifest__.py (hunk 15 above)
Comment (kmagusiak) [reply addons/base_automation/__manifest__.py:18]: If you don't want a bridge module, why not add `base_automation` as a dependency to `digest` and move the code there?

The `base_automation` module, which is managed by the framework team, *should not* depend on modules by other teams. 
BTW: In the past at a customer, we had disgest deinstalled, but we were using base_automation.

Comment (amdi-odoo) [reply addons/base_automation/__manifest__.py:18]: If we add `base_automation` as a dependency to `digest`, as soon as we install a module that depends on `digest` then `base_automation` will automatically be installed. 
Will that be an issue ? :thinking:

Comment (tde-banana-odoo) [reply addons/base_automation/__manifest__.py:18]: Well digest is considered as a base feature. You can deactivate it by removing digests. It is easier than having lot of bridge modules. I think digest is a technical module that could even be in mail, base automation is a functional module (for technical people), so it makes sense to have digest in dependencies of base automation, not the reverse move.

Comment (tde-banana-odoo) [reply addons/base_automation/__manifest__.py:18]: ``The base_automation module, which is managed by the framework team, should not depend on modules by other teams.``

You mean like mail (us), sms (us) and resources (vidange) :D ?

Comment (kmagusiak) [reply addons/base_automation/__manifest__.py:18]: I was only considering mail as a base feature, not sms or digest. Anyways, it's ok like you suggest because we already have these other modules here.

Comment (tde-banana-odoo) [reply addons/base_automation/__manifest__.py:18]: In a perfect world I would make bridge modules, but those have a cost and we are told to try to avoid them as possible, hence adding "fake" dependencies ... what matter is that you can de-activate the digest feature by simply killing digest records.

**Comment by tde-banana-odoo** [comment]
@robodoo r+
